### DISCUSSION TITLE
Path segments with spaces (%20)

### DISCUSSION THREAD

**Discussion by svoop** [jeremyevans/roda#292] [General]
I'm porting a simple tool for aviation [AIPHub](https://aiphub.tower.zone) from Sinatra to Roda. One feature is [translating coordinates](https://aiphub.tower.zone/point). Doing so from DMS to DD means the URL contains spaces encoded as `%20`:

https://aiphub.tower.zone/point/48%2051%2031,2%2017%2040

Simple enough in the router:

```ruby
class Application < Roda
  hash_branch('point') do |r|
    r.get String do |coordinates|
      debugger
    end
  end
end
```

Works just fine, but `coordinates` contains the URL undecoded, the `%20` are still there. Is this the expected behaviour? No biggie decoding it, but I'm surprised anyway.

Since Roda is slim by default, I've checked the plugins and found `unescape_path`. However, it seems to be made for a different use case, at least with the plugin enabled and `%20` in the URL, the above route is never hit anymore.

**Comment by jeremyevans** [comment]
I think you are hitting the same bug as #286.  See workaround in that issue.  This will be fixed in 3.62.0, which will be released on Monday.

**Reply by svoop** [reply]
Oh, I didn't connect the dots when I skimmed over the issues, thanks a bunch for the hint – and for the fix!

**Reply by svoop** [reply]
@jeremyevans Thanks a lot for the release, works like a charm now!
